From 1799926f0af4b9e276158ef5f0103a75eca8265b Mon Sep 17 00:00:00 2001
From: Prasanna S <prasannas@ami.com>
Date: Tue, 22 Apr 2025 11:57:39 +0530
Subject: [PATCH 1/2] ARM FVP:Firmware Update Support

This commit adds support for the ARM WIC partition in the
obmc-flash-bmc file & Fixed Updated Partition Switch Issue

Upstream-Status: Inappropriate

Signed-off-by: Prasanna S <prasannas@ami.com>
Signed-off-by: Meghana G <meghanag@ami.com>
Signed-off-by: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
---
 bmc/obmc-flash-bmc | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/bmc/obmc-flash-bmc b/bmc/obmc-flash-bmc
index f9cc4e25..dbb19be9 100644
--- a/bmc/obmc-flash-bmc
+++ b/bmc/obmc-flash-bmc
@@ -576,13 +576,11 @@ function mmc_mount() {

 function mmc_update() {
     # Update u-boot if needed
-    bootPartition="mmcblk0boot0"
+    bootPartition="mmcblk0p1"
     devUBoot="/dev/${bootPartition}"
     imgUBoot="${imgpath}/${version}/image-u-boot"
     if [ "$(cmp_uboot "${devUBoot}" "${imgUBoot}")" != "0" ]; then
-        echo 0 > "/sys/block/${bootPartition}/force_ro"
         dd if="${imgUBoot}" of="${devUBoot}"
-        echo 1 > "/sys/block/${bootPartition}/force_ro"
     fi

     # Update the secondary (non-running) boot and rofs partitions.
--
2.43.0

